### ISSUE TITLE
scylla won't start if hints directory not available

### ISSUE THREAD

**Issue by glommer** [scylladb/scylla#3514]
@psarna I thought we fixed this ?

Using current HEAD, d91891b6f0062269f4140b98eef3afcadb409fa5

I get :
```
ERROR 2018-06-13 09:49:59,273 [shard 0] seastar - Exiting on unhandled exception: boost::filesystem::filesystem_error (boost::filesystem::space: No such file or directory: "/var/lib/scylla/0/hints")
```

This comes from:

```
734	    db::hints::resource_manager::max_shard_disk_space_size = boost::filesystem::space(dbcfg.hints_directory().c_str()).capacity / (10 * smp::count);
```

All directories should be created by scylla if not present. We do that for data and commitlog already.

**Comment by psarna** [comment]
Still in review: https://github.com/psarna/scylla/commit/d962395f3fc4b178c556c94c1ce13cbb7db4b91e and https://groups.google.com/forum/#!searchin/scylladb-dev/max_shard_disk_space_size|sort:date/scylladb-dev/BwzLo_2wAn8/xiNltGdWAgAJ

It's a regression after `hinted_handoff` was made optional again, but storage_proxy still tried to use its directory as an initializer. It's also the cause of #3495

**Comment by duarten** [comment]
Fixed by 6b3a97e34a37e23ef2ba47effa0486dfddad54b4.
